tools/mmp/driver.cs

```diff
@@ -1685,8 +1712,19 @@ static void ProcessNativeLibrary (HashSet<string> processed, string library, Lis
 				// install_name_tool gets angry if you copy in a read only native library
 				CopyFileAndRemoveReadOnly (real_src, dest);
 			}
-
+			
 			bool isStaticLib = real_src.EndsWith (".a", StringComparison.Ordinal);
+			bool isDynamicLib = real_src.EndsWith (".dylib", StringComparison.Ordinal);
+
+			if (isDynamicLib && NativeStrip == NativeStripMode.Strip) {
+				string archToRemove = arch == "i386" ? "x86_64" : "i386";
+				int ret = XcodeRun ("lipo", $"{dest} -remove {archToRemove} -output {dest}");
+				if (ret != 0)
+					throw new MonoMacException (5310, true, "lipo failed with an error code '{0}'. Check build log for details.", ret);
```

Comment:
is this a new error? if so it should be documented

Reply:
Yes, I forgot.